Cabal-tests/tests/UnitTests/Distribution/Simple/Glob.hs

```diff
@@ -53,23 +53,26 @@ makeSampleFiles dir = for_ sampleFileNames $ \filename -> do
 compatibilityTests :: CabalSpecVersion -> [TestTree]
 compatibilityTests version =
   [ testCase "literal match" $
-      testMatches "foo/a" [GlobMatch "foo/a"]
-  , testCase "literal no match on prefix" $
-      testMatches "foo/c.html" [GlobMatchesDirectory "foo/c.html"]
+      testMatches "foo/a" (Left "foo/a")
+      -- TODO: This is kind of confusing - I feel like someone writing that they
+      -- expect a file `foo/c.html` in `extra-source-files` should probably
+      -- get a Problem if the match in question is `foo/c.html/blah` ?
+--   , testCase "literal no match on prefix" $
+--       testMatches "foo/c.html" (Left "foo/c.html")
```

Comment:
This seems odd, yeah. I'd support making it an error, although it would be really nice if we had a [`crater`](https://github.com/rust-lang/crater)-like tool we could use to check the impact of a change like this.